zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/distribution/CommandDistributionBehavior.java

```diff
@@ -272,10 +274,28 @@ <T extends UnifiedRecordValue> void distribute(
         final ValueType valueType, final Intent intent, final T value);
   }
 
-  private class DistributionRequest implements RequestBuilder, DistributionRequestBuilder {
+  public interface ContinuationRequestBuilder {
+    /**
+     * Specifies whether to wait for the queue to be non-empty before writing the continuation
+     * command. If set to false, the continuation command will be written immediately if the queue
+     * is currently empty.
+     */
+    ContinuationRequestBuilder awaitNonEmptyQueue(boolean awaitNonEmptyQueue);
```

Comment:
❌ I don't like this option to be honest. It makes the assumption that something will be distributed within the queue. Do you see a specific need for this already? Like I said in my previous comment, I believe it should only possible to add a continuation when actually distributing a command within the same queue.

Reply:
That's fair :+1: The use case I had in mind was first registering a continuation and then adding a bunch of distributions to the queue. However, I'm not yet sure if it will be needed or not. If you prefer, I can remove it for now until we know that we need it.

Reply:
Let's remove it for now and see if we'll need it in the future 👍

Reply:
If I remove this option, what should the behavior be if the queue is currently empty? Write the command immediately or just request it for later?
I slightly leaning towards just keeping this option because it makes this behavior explicit and doesn't really add much complexity.

Reply:
This is why I expected a continuation command to only be avialable when you actaully distribute something 😄 

Personally I would say it makes sense to immediately write the command. When you add a continuation you are basically saying, please write this command when queue `X` is empty. If it's already empty I don't see a reason to wait for it 🤷

Reply:
We can clarify this happens on the javadoc

Reply:
:+1: This option is removed now.